Server/controllers/user.controller.js

```diff
@@ -1,17 +1,73 @@
-"use strict";
+const userDataLayer = require('../datalayer/user.datalayer');
 /**
  * Class representing users controller
  *
  * @class usersController
  */
 class usersController {
 
-  static createUser (req, res) {}
+  static async createUser (req, res) {
+    const userObj = {
+      username : req.body.username,
+      email : req.body.email,
+      password : req.body.password,
+      firstname : req.body.firstname,
+      lastname : req.body.lastname
+    }
+    try {
+        await userDataLayer.createUser(userObj);
+    } catch(err) {
+        res.status(400).send(err.message);
+        return;
+    }
+    res.status(200).json({"message" : "User created successfully"});
+  }
 
-  static getUsersByUserId (req, res) {}
 
-  static updateUserById (req, res) {}
+  static async getUserById (req, res) {
+    let userid = req.params._id;
+    try {
+      user = await userDataLayer.getUserById(userid);
+    }
+    catch(err) {
+      res.status(400).send(err.message);
+      return;
+    }
+    res.status(200).json(user);
+  }
 
-  static deleteUserById (req, res) {}
+
+  static async updateUserById (req, res) {
+    let userid = req.params._id;
+    const updateUserObj = {
+      username : req.body.username,
+      email : req.body.email,
+      password : req.body.password,
+      firstname : req.body.firstname,
+      lastname : req.body.lastname
```

Comment:
Allow support of updating a single field say just firstname

Reply:
Added support for updating single field also